File: rules_intellij/testing/src/com/google/idea/testing/BazelTestApplication.java
/*
 * This file is based on Bazel plugin for IntelliJ by The Bazel Authors, licensed under Apache-2.0;
 * It was modified by JetBrains s.r.o. and contributors
 */
package com.google.idea.testing;

import org.jetbrains.annotations.TestOnly;
import org.junit.jupiter.api.extension.ExtendWith;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@TestOnly
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@ExtendWith(BazelTestApplicationExtension.class)
public @interface BazelTestApplication {}

File: pluginTests/testFixtures/org/jetbrains/bazel/workspace/model/test/framework/BazelTestApplication.kt
package org.jetbrains.bazel.workspace.model.test.framework

import com.intellij.testFramework.junit5.fixture.TestFixtures
import com.intellij.testFramework.junit5.impl.TestApplicationExtension
import org.jetbrains.annotations.TestOnly
import org.junit.jupiter.api.extension.ExtendWith

@TestOnly
@Target(AnnotationTarget.CLASS)
@ExtendWith(
  TestApplicationExtension::class, BazelIdeaTextExtension::class,
)
@TestFixtures
annotation class BazelTestApplication
